File: back-end/seed.js
// src/utils/seed.js
// Run: node src/utils/seed.js
require("dotenv").config();
const mongoose    = require("mongoose");
const User        = require("../models/User.model");
const Account     = require("../models/Account.model");
const Transaction = require("../models/Transaction.model");
const Budget      = require("../models/Budget.model");
const Goal        = require("../models/Goal.model");

async function seed() {
  await mongoose.connect(process.env.MONGO_URI);
  console.log("🌱  Seeding database…");

  // Clean up
  await Promise.all([
    User.deleteMany({}), Account.deleteMany({}),
    Transaction.deleteMany({}), Budget.deleteMany({}), Goal.deleteMany({}),
  ]);

  // Create demo user
  const user = await User.create({
    name: "Arjun Kumar",
    email: "arjun@example.com",
    password: "password123",
    phone: "+91 9876543210",
    plan: "premium",
  });
  console.log(`✅  User created: ${user.email}`);

  // Accounts
  await Account.insertMany([
    { user: user._id, name: "HDFC Savings Account",  type: "Savings",     balance: 142350,  last4: "4821", color: "#2563eb", icon: "building2" },
    { user: user._id, name: "SBI Salary Account",    type: "Salary",      balance: 68200,   last4: "3309", color: "#16a34a", icon: "wallet"    },
    { user: user._id, name: "ICICI Credit Card",     type: "Credit Card", balance: -23450,  last4: "9977", color: "#dc2626", icon: "credit-card"},
    { user: user._id, name: "Zerodha Portfolio",     type: "Investment",  balance: 315000,  last4: "—",    color: "#7c3aed", icon: "line-chart" },
  ]);
  console.log("✅  Accounts seeded");

  // Transactions (last 30 days sample)
  const payeesByCat = {
    food:      ["Zomato","Swiggy","McDonald's","Pizza Hut","Café Coffee Day"],
    transport: ["Uber","Ola","IndiGo","Indian Railways","Indian Oil"],
    shopping:  ["Amazon India","Flipkart","Myntra","DMart","Croma"],
    utilities: ["TSSPDCL","Airtel","Jio Recharge","Hyderabad Water Board"],
    health:    ["Apollo Pharmacy","Practo","Cult.fit","MedPlus"],
    entertain: ["Netflix","Spotify","BookMyShow","Disney+ Hotstar"],
    savings:   ["SIP – Mirae Asset","PPF – Post Office","RD – SBI"],
    income:    ["Salary – TCS","Freelance Payment","Dividend – HDFC"],
  };
  const cats = Object.keys(payeesByCat);
  const txns = [];
  for (let i = 0; i < 60; i++) {
    const d = new Date(); d.setDate(d.getDate() - i);
    const isIncome = Math.random() < 0.07;
    const cat  = isIncome ? "income" : cats.filter(c => c !== "income")[Math.floor(Math.random() * 7)];
    const list = payeesByCat[cat];
    txns.push({
      user:     user._id,
      payee:    list[Math.floor(Math.random() * list.length)],
      amount:   isIncome ? Math.round(Math.random() * 40000 + 5000) : Math.round(Math.random() * 3000 + 50),
      type:     isIncome ? "credit" : "debit",
      category: cat,
      date:     d,
    });
  }
  // Add monthly salary
  txns.push({ user: user._id, payee: "Salary – TCS", amount: 85000, type: "credit", category: "income", date: new Date() });
  await Transaction.insertMany(txns);
  console.log(`✅  ${txns.length} transactions seeded`);

  // Budgets
  const month = new Date().toISOString().slice(0, 7);
  await Budget.insertMany([
    { user: user._id, category: "food",      limit: 8000,  month },
    { user: user._id, category: "transport", limit: 3000,  month },
    { user: user._id, category: "shopping",  limit: 5000,  month },
    { user: user._id, category: "entertain", limit: 2000,  month },
    { user: user._id, category: "utilities", limit: 4000,  month },
    { user: user._id, category: "health",    limit: 3000,  month },
  ]);
  console.log("✅  Budgets seeded");

  // Goals
  await Goal.insertMany([
    { user: user._id, name: "Emergency Fund",    target: 300000,  saved: 142000, icon: "shield",    color: "#2563eb", deadline: new Date("2025-12-31") },
    { user: user._id, name: "Europe Vacation",   target: 150000,  saved: 42000,  icon: "award",     color: "#16a34a", deadline: new Date("2026-06-30") },
    { user: user._id, name: "MacBook Pro",        target: 180000,  saved: 90000,  icon: "laptop",    color: "#7c3aed", deadline: new Date("2025-09-30") },
    { user: user._id, name: "Home Down Payment",  target: 1000000, saved: 310000, icon: "home",      color: "#d97706", deadline: new Date("2028-01-01") },
  ]);
  console.log("✅  Goals seeded");

  console.log("\n🎉  Seed complete!");
  console.log(`   Login: arjun@example.com / password123`);
  mongoose.disconnect();
}

seed().catch(err => { console.error(err); process.exit(1); });

File: back-end/seed.py
"""
Run once to populate demo data:
    python seed.py
"""
import sys, os
sys.path.insert(0, os.path.dirname(__file__))

from database import init_db, get_conn
from auth_utils import hash_password
from datetime import datetime, timedelta
import random

PAYEES = {
    "food":      ["Zomato","Swiggy","Café Coffee Day","McDonald's","Dosa Corner","Pizza Hut","Barbeque Nation"],
    "transport": ["Uber","Ola","HMRL Metro","Indian Oil","Rapido","IndiGo Airlines","Indian Railways"],
    "housing":   ["Rent – Landlord","Society Maintenance","Home Loan EMI – SBI","Gas Cylinder – HP"],
    "health":    ["Apollo Pharmacy","Practo Consult","Cult.fit Membership","MedPlus","Star Health Insurance"],
    "shopping":  ["Amazon India","Flipkart","Myntra","DMart","Reliance Trends","Nykaa","Croma"],
    "entertain": ["Netflix","Spotify Premium","BookMyShow","Steam","Disney+ Hotstar","YouTube Premium"],
    "savings":   ["SIP – Mirae Asset","RD – SBI","PPF – Post Office","NPS Contribution"],
    "utilities": ["TSSPDCL Electricity","Airtel Broadband","Hyderabad Water Board","Jio Recharge"],
    "other":     ["ATM Withdrawal","Miscellaneous","Bank Charges – HDFC"],
}
CATEGORIES = list(PAYEES.keys())

def seed():
    init_db()
    conn = get_conn()

    # Check if already seeded
    existing = conn.execute("SELECT id FROM users WHERE email='arjun@example.com'").fetchone()
    if existing:
        print("⚠️  Demo data already exists. Skipping seed.")
        conn.close()
        return

    # Create demo user
    user_id = conn.execute(
        "INSERT INTO users (name, email, phone, password, plan) VALUES (?,?,?,?,?)",
        ("Arjun Kumar", "arjun@example.com", "+91 9876543210", hash_password("password123"), "premium")
    ).lastrowid
    print(f"✅ Created user: arjun@example.com / password123 (id={user_id})")

    # Generate 90 days of transactions
    now  = datetime.now()
    txns = []

    # Monthly salary for 3 months
    for m in range(3):
        d = (now.replace(day=1) - timedelta(days=m*28)).replace(day=1)
        txns.append((user_id, d.strftime("%Y-%m-%d"), "Salary – TCS", "income", 85000.0, "credit", "Monthly salary"))

    # Random transactions
    for i in range(90):
        d   = now - timedelta(days=i)
        cat = random.choice(CATEGORIES)
        pay = random.choice(PAYEES[cat])
        amt = round(random.uniform(50, 3000), 2)
        txns.append((user_id, d.strftime("%Y-%m-%d"), pay, cat, amt, "debit", ""))

    conn.executemany(
        "INSERT INTO transactions (user_id,date,payee,category,amount,type,note) VALUES (?,?,?,?,?,?,?)",
        txns
    )
    print(f"✅ Created {len(txns)} transactions")

    # Budgets for current month
    month = now.strftime("%Y-%m")
    budgets = [
        (user_id, "food",      month, 8000),
        (user_id, "transport", month, 3000),
        (user_id, "shopping",  month, 5000),
        (user_id, "entertain", month, 2000),
        (user_id, "utilities", month, 4000),
        (user_id, "health",    month, 3000),
    ]
    conn.executemany("INSERT INTO budgets (user_id,category,month,lim) VALUES (?,?,?,?)", budgets)
    print(f"✅ Created {len(budgets)} budgets")

    # Goals
    goals = [
        (user_id, "Emergency Fund",    300000, 142000, "2025-12-31", "#2563eb"),
        (user_id, "Europe Vacation",   150000, 42000,  "2026-06-30", "#16a34a"),
        (user_id, "MacBook Pro",       180000, 90000,  "2025-09-30", "#7c3aed"),
        (user_id, "Home Down Payment", 1000000, 310000,"2028-01-01", "#d97706"),
    ]
    conn.executemany("INSERT INTO goals (user_id,name,target,saved,deadline,color) VALUES (?,?,?,?,?,?)", goals)
    print(f"✅ Created {len(goals)} goals")

    conn.commit()
    conn.close()
    print("\n🎉 Seed complete! Login: arjun@example.com / password123")

if __name__ == "__main__":
    seed()
